docker/defaults.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+
+	"github.com/docker/docker/api"
+)
+
+var (
+	configFileValues  map[string]interface{}
+	defaultStorageDir string
+	defaultGetter     = DefaultGetter{}
+	dockerConfigPath  string
+	dockerCertPath    string
+	dockerTlsVerify   bool
+	configFileHost    string
+)
+
+type DefaultGetter struct {
+	configFileValues map[string]interface{}
+}
+
+type ConfigHierarchy struct {
+	EnvVar  string
+	JsonKey string
+	Default interface{}
+}
+
+// The hierarchy of configuration options flows like
+// this, in order of most preferred to least preferred:
+//
+// Command Line Flags => Environment Variables => defaults.json => Hardcoded Defaults
+func getHomeDir() string {
+	if runtime.GOOS == "windows" {
+		return os.Getenv("USERPROFILE")
+	}
+	return os.Getenv("HOME")
+}
+
+func getConfigPath() string {
+	// This is a "special case" since we
+	// need to know about it to check the other
+	// defaut values.
+	envVal := os.Getenv("DOCKER_CONFIG_PATH")
```

Comment:
Yes!  We'll need a way to allow more than one docker CLI to run at a time and each may need its own config info.  Another option is to allow the config file location to be passed on the cli cmd line, but we can decide on that later.  Having this env var for now is good enough.